app/assets/javascripts/discourse/app/components/sidebar/api-section.js

```diff
@@ -1,10 +1,47 @@
 import Component from "@glimmer/component";
 import { getOwner, setOwner } from "@ember/application";
+import { inject as service } from "@ember/service";
 
 export default class SidebarApiSection extends Component {
+  @service sidebarState;
+
   constructor() {
     super(...arguments);
     this.section = new this.args.sectionConfig();
     setOwner(this.section, getOwner(this));
   }
+
+  get shouldDisplay() {
+    if (!this.sidebarState.currentPanel.filterable) {
+      return true;
+    }
+    const shouldDisplay =
+      this.sidebarState.filter.length === 0 || this.filteredLinks.length > 0;
+    const index = this.sidebarState.filteredOutSections.indexOf(
+      this.section.name
+    );
+    if (shouldDisplay) {
+      if (index !== -1) {
+        this.sidebarState.filteredOutSections.removeObject(this.section.name);
```

Comment:
This and addObject are causing computation errors in ember:

![image](https://github.com/discourse/discourse/assets/920448/f47d782a-010b-4b7b-8b5e-84a89997abbd)

None of this is actually necessary, since ember recomputes `filteredLinks` everytime `sidebarState.filter` changes, and this `shouldDisplay` also recomputes every time  `sidebarState.filter` or `this.filteredLinks` changes :) You can remove this code and just return `shouldDisplay`.